src/methods/tx/_events.ts

```diff
@@ -0,0 +1,345 @@
+        /**
+         * parsedData: <Array> [AccountId, IntentionType, IntentionID, Balance, Balance]
+         *                     [who, intention type, intention id, amount, amount sold/bought]
+         */
+        if (Array.isArray(parsedData)) {
+          mergeEventToScope({
+            ...exchangeTxEventData,
+            data: {
+              id: parsedData[2]?.toString(),
+              intentionType: parsedData[1]?.toString(),
+              account: parsedData[0]?.toString(),
+              amountAmmTrade: new BigNumber(parsedData[3]?.toString() || 0),
+              amountOutAmmTrade: new BigNumber(parsedData[4]?.toString() || 0),
+            },
+          });
+        }
+        break;
+      case 'IntentionResolvedDirectTrade':
+        /**
+         * parsedData: <Array> [AccountId, AccountId, IntentionID, IntentionID, Balance, Balance]
+         *                     [User1 acc id, User2 acc id, intention id 1, intention id 2, amount 1, amount 2]
+         *
+         * First amount is amount of asset A going from first account to second account,
+         * and the second amount is asset B going from second account to first account.
+         *
+         * Which assets have been used - check in event "IntentionRegistered" by
+         * appropriate IntentionID.
+         *
+         * One exchange action (sell/buy) can includes multiple direct trade transactions,
+         * that's why we need track all direct trade transactions for one exchange action.
+         */
+        if (Array.isArray(parsedData)) {
+          mergeEventToScope({
+            ...exchangeTxEventData,
+            intentions: [parsedData[2]?.toString(), parsedData[3]?.toString()],
+            data: {
+              id: parsedData[2]?.toString(),
+              directTrades: [
+                {
+                  amountSent: new BigNumber(parsedData[4]?.toString() || 0),
+                  amountReceived: new BigNumber(parsedData[5]?.toString() || 0),
+                  account1: parsedData[0]?.toString(),
+                  account2: parsedData[1]?.toString(),
+                  pairedIntention: parsedData[3]?.toString(),
+                },
+              ],
+            },
+          });
+          mergeEventToScope({
+            ...exchangeTxEventData,
+            intentions: [parsedData[2]?.toString(), parsedData[3]?.toString()],
+            data: {
+              id: parsedData[3]?.toString(),
+              directTrades: [
+                {
+                  amountSent: new BigNumber(parsedData[5]?.toString() || 0),
+                  amountReceived: new BigNumber(parsedData[4]?.toString() || 0),
+                  account1: parsedData[1]?.toString(),
+                  account2: parsedData[0]?.toString(),
+                  pairedIntention: parsedData[2]?.toString(),
+                },
+              ],
+            },
+          });
+        }
+        break;
+      case 'IntentionResolvedDirectTradeFees':
```

Comment:
Is it intentional to leave this case blank?

Reply:
Implementation of this event is still not completed on node side but we need process data from it so I leaved placeholder.